### PR TITLE
[DomCrawler] Abstract URI logic and crawl images

### PR DESCRIPTION
**Pull request by valeriangalliat**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #12429 |
| License | MIT |
| Doc PR | symfony/symfony-docs#4971 |

### CODE REVIEW

**Comment by valeriangalliat** [comment]
The DomCrawler tests are passing, but there's a problem with FrameworkBundle. I don't think it's caused by my changes since I have the same failure on the current `master`.

**Comment by WouterJ** [comment]
This can easily made BC by still letting `Form` extend `Link`. This means both implement the interface and both extend the abstract class. This can then be merged into Symfony 2.7 and the `extend Link` can be removed in Symfony 3.0.

This would allow users to already make their code compatible for Symfony 3.0.

**Comment by valeriangalliat** [comment]
@WouterJ I just fixed this in a separate commit, so it's easy to revert in Symfony 3.0. Do I close this PR and make another one targeting the 2.7 branch?

**Comment by stof** [comment]
@valeriangalliat please send a PR to 2.7 yes. It is better to ship new features in 2.7 rather than making them wait until November for 3.0. Once it is merged, you would then send a second PR with the 3.0 cleanup.

File: src/Symfony/Component/DomCrawler/AbstractUriElement.php [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+{
+    /**
+     * @var \DOMElement
+     */
+    protected $node;
+
+    /**
+     * @var string The method to use for the element URI
+     */
+    protected $method;
```
Comment (stof) [inline src/Symfony/Component/DomCrawler/AbstractUriElement.php]: please make these private (we have public getters for them anyway)

Comment (stof) [inline src/Symfony/Component/DomCrawler/AbstractUriElement.php]: hmm, no. These properties are currently protected in Link, so we cannot make them private yet

**Pull request closed without merging**